packages/ui/src/Popup/ImportLedger/ImportLedger.tsx

```diff
@@ -56,6 +60,36 @@ function ImportLedger (): React.ReactElement {
     }
   }, [address]);
 
+  // Set accountIndex and addressOffset to next available pair
+  useEffect(() => {
+    const ledgerAccounts = accounts.filter((account) => account.isHardware && account.hardwareType === 'ledger');
+    const existingIndexOffsetMap = ledgerAccounts.reduce((indexOffsetMap: number[][], account) => {
+      const index = account.accountIndex as number;
+      const offset = account.addressOffset as number;
+
+      if (indexOffsetMap[index]) {
+        indexOffsetMap[index].push(offset);
+      } else {
+        indexOffsetMap[index] = [offset];
+      }
+
+      return indexOffsetMap;
+    }, []);
```

Comment:
This creates a map of the index/offset pairs like this:

```
{ index: offset[] }

e.g.

{
  0: [0, 1, 3],
  3: [1, 2]
}
```

This is used to look up if a certain pair is taken or not.